landlab/components/flow_accum/flow_accumulator.py

```diff
@@ -957,15 +959,27 @@ def _add_director(self, flow_director):
 
     def _add_depression_finder(self, depression_finder):
         """Test and add the depression finder component."""
-        PERMITTED_DEPRESSION_FINDERS = ["DepressionFinderAndRouter"]
+        PERMITTED_DEPRESSION_FINDERS = ["DepressionFinderAndRouter", "LakeMapperBarnes"]
 
         # now do a similar thing for the depression finder.
         self._depression_finder_provided = depression_finder
         if self._depression_finder_provided is not None:
 
             # collect potential kwargs to pass to depression_finder
             # instantiation
-            potential_kwargs = ["routing"]
+            potential_kwargs = [
+                "routing",
+                "pits",
+                "reroute_flow",
+                "surface",
+                "method",
+                "fill_flat",
+                "fill_surface",
+                "redirect_flow_steepest_descent",
+                "reaccumulate_flow",
+                "ignore_overfill",
+                "track_lakes",
+            ]
```

Comment:
Would [__slots__](https://docs.python.org/3/reference/datamodel.html#slots) be useful here? This is a speculative comment, not a review comment.